kyo-data/shared/src/main/scala/kyo/internal/ForSome.scala

```diff
@@ -0,0 +1,52 @@
+package kyo.internal
+
+/** Existentinal types encoding for type constructors `F[_]`. `ForSome[F]` is semantically equiavalent to `F[?]`
+  */
+type ForSome[F[_]] = ForSome.Type[F]
```

Comment:
Given that existential types were remove due to unsoundness, I wonder if we could have issues with this mechanism. I think it's ok to merge and see how it goes, though

Reply:
I have fixed the encoding, now it doesn't use `asInstanceOf`.
However, now there is no guarantee that no extra allocation (of `Unwrap`) can happen.